### PR TITLE
Credential rotation cmd

### PR DESCRIPTION
**Pull request by luizbranco**

### CODE REVIEW

File: specs/connector.yaml [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+        200:
+          description: The created OAuth 2.0 credential pair.
+          schema: { $ref: '#/definitions/OAuthCredentialCreateResponse' }
+        400: { $ref: '#/responses/BadRequest' }
+        401: { $ref: '#/responses/Unauthorized' }
+        404: { $ref: '#/responses/NotFound' }
+        500: { $ref: '#/responses/Internal' }
+
+  /sso:
+    x-manifold-audience: browser
```
Comment (jbowes) [inline specs/connector.yaml]: we filter the spec files based on annotations like this. you can run `tools/bin/openapi-preprocessor --audience=provider specs/connector.yaml connector.yaml` to get the version suitable for use in this repo.

File: cmd/credentials.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+	fmt.Println("Your old credentials will expire automatically in 24 hours")
+	fmt.Println("Make sure to copy your new credentials now. You won’t be able to see this information again!")
+	fmt.Printf("Client ID: %s\n", payload.ID)
+	fmt.Printf("Secret: %s\n", payload.Secret)
+
+	return nil
+}
+
+func NewConnector(token string) (*client.Connector, error) {
```
Comment (jbowes) [inline cmd/credentials.go]: FWIW we have a placeholder repo at https://github.com/manifoldco/go-connector that we could use for an oag generated client, eventually.

File: cmd/credentials.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+const apiURL = "https://api.%s.manifold.co/v1"
+
+func init() {
+	cmd := cli.Command{
+		Name:  "credentials",
+		Usage: "Rotates OAuth credentials for Manifold.co",
+		Subcommands: []cli.Command{
+			{
+				Name:   "rotate",
+				Usage:  "Creates a new credential and set the old one to expire in 24h",
```
Comment (jbowes) [inline cmd/credentials.go]: "sets the old one"

File: cmd/credentials.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+	if providerLabel == "" {
+		return cli.NewExitError("--provider flag missing", -1)
+	}
+
+	productLabel := cliCtx.String("product")
+	if productLabel == "" {
+		return cli.NewExitError("--product flag missing", -1)
+	}
+
+	p := promptui.Prompt{
```
Comment (jbowes) [inline cmd/credentials.go]: I think you should include a message that says the user is logging in to their manifold account

File: cmd/credentials.go [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+		return cli.NewExitError(err.Error(), -1)
+	}
+
+	cfgs = append(cfgs, manifold.WithAPIToken(token))
+	client = manifold.New(cfgs...)
+
+	var provider *manifold.Provider
+	var product *manifold.Product
+
+	provList := client.Providers.List(ctx)
```
Comment (jbowes) [inline cmd/credentials.go]: `defer provList.Close()`

File: cmd/credentials.go [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+			provider = p
+			break
+		}
+	}
+
+	if provider == nil {
+		return cli.NewExitError(fmt.Sprintf("Provider %q not found", providerLabel), -1)
+	}
+
+	prodList := client.Products.List(ctx, nil)
```
Comment (jbowes) [inline cmd/credentials.go]: you can give the provider id in the list opts

Comment (luizbranco) [reply cmd/credentials.go]: Oh right, I had that at first when I messing around with the ref Label, I ended up deleting all options. Will add again.

File: cmd/credentials.go (hunk 2 above)
Comment (luizbranco) [reply cmd/credentials.go]: Should I move the code there instead? I was planning to use `go-manifold`, but hit some roadblocks with oag.

Comment (jbowes) [reply cmd/credentials.go]: I'd stick with what you're doing right now, and once oag can handle allof and refs, we can put this stuff in go-connector, and update grafton to use it.

**Pull request merged**